src/github.com/matrix-org/dendrite/syncapi/storage/syncserver.go

```diff
@@ -310,6 +310,11 @@ func (d *SyncServerDatabase) CompleteSync(
 
 		stateEvents = removeDuplicates(stateEvents, recentEvents)
 		jr := types.NewJoinResponse()
+		if prevBatch := recentStreamEvents[0].streamPosition - 1; prevBatch > 0 {
+			jr.Timeline.PrevBatch = types.StreamPosition(prevBatch).String()
+		} else {
+			jr.Timeline.PrevBatch = recentStreamEvents[0].streamPosition.String()
```

Comment:
It's probably clearer to explicitly set this to `types.StreamPosition(0).String()` or whatever, otherwise it feels a bit confusing.

Reply:
Fixed in https://github.com/matrix-org/dendrite/pull/589/commits/ac0a928702853a17b36c793f44c778b6d883ebda. I used `types.StreamPosition(0).String()` as the stream position sequence starts at 1.